### PR TITLE
Adds polygon nodes and enables query of polygon balances

### PR DESCRIPTION
**Pull request by nebolax**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

**Review by yabirgb — COMMENTED** [review]
There is a failing test and this is lacking tests at least for the db upgrade. For the balances don't we have tests in optimism?

File: rotkehlchen/constants/assets.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
 A_USDT = Asset('eip155:1/erc20:0xdAC17F958D2ee523a2206206994597C13D831ec7')
 A_USDC = Asset('eip155:1/erc20:0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48')
 A_TUSD = Asset('eip155:1/erc20:0x0000000000085d4780B73119b644AE5ecd22b376')
+A_MATIC = Asset('MATIC')
```
[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/constants/assets.py:33]: Yup, thanks. Forgot that I have put the erc20 token in some places already. I also think that having asset with identifier `MATIC` is better

File: rotkehlchen/db/upgrades/v37_v38.py [hunk 2]
```diff
@@ -17,6 +17,22 @@ def _add_polygon_pos_location(write_cursor: 'DBCursor') -> None:
     write_cursor.execute('INSERT OR IGNORE INTO location(location, seq) VALUES ("h", 40);')
 
 
+def _add_polygon_pos_nodes(write_cursor: 'DBCursor') -> None:
+    default_polygon_pos_rpc_nodes = [
+        ('polygon etherscan', '', 0, 1, 0.25, 'POLYGON_POS'),
+        ('ankr', 'https://rpc.ankr.com/polygon', 0, 1, 0.15, 'POLYGON_POS'),
```
[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/db/upgrades/v37_v38.py:23]: I checked their website and don't know tbh. They are both free and look similar, so I think either is fine

File: rotkehlchen/db/upgrades/v37_v38.py [hunk 3]
```diff
@@ -17,6 +17,22 @@ def _add_polygon_pos_location(write_cursor: 'DBCursor') -> None:
...
+        ('BlockPi', 'https://polygon.blockpi.network/v1/rpc/public', 0, 1, 0.15, 'POLYGON_POS'),
```
[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/db/upgrades/v37_v38.py:24]: I took the free nodes from rpclist.com

> Do you have a comparasion of which one is better in terms of speed and privacy?

rpclist.com for some reason is down right now, but as far as I remember all the nodes had relatively same characteristics. In terms of privacy, how would you suggest to check that?

> Also this one offers a full node in the free tier. Have you checked the others?

Afaik defillama offers an archieve node too, but I haven't checked. Does it matter for the upgrade though? Nodes capabilities are determined on the fly anyway

File: rotkehlchen/data/global.db
[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/data/global.db:1]: Yup, here it is
```sql
INSERT INTO assets(rowid,identifier,name,type) VALUES(5075,'MATIC','MATIC','B');
INSERT INTO common_asset_details(rowid,identifier,symbol,coingecko,cryptocompare,forked,started,swapped_for) VALUES(5075,'MATIC','MATIC','matic-network','MATIC',NULL,NULL,NULL);
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(11,'polygon etherscan','',0,1,'0.25','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(12,'ankr','https://rpc.ankr.com/polygon',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(13,'BlockPi','https://polygon.blockpi.network/v1/rpc/public',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(14,'PublicNode','https://polygon-bor.publicnode.com',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(15,'DefiLlama','https://polygon.llamarpc.com',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(16,'1rpc','https://1rpc.io/matic',0,1,'0.15','POLYGON_POS');
```

File: rotkehlchen/db/upgrades/v37_v38.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/db/upgrades/v37_v38.py:24]: In order to check the latency of the nodes I looked at https://chainlist.org/chain/137.

By monitoring it for a bit I can say that nodes' latency jumps up and down, so it's not something really stable. So I would say that it will be okay to have same priority of all the nodes.

File: rotkehlchen/data/global.db
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/data/global.db:1]: I tough we had this token already. Maybe is fine to add it but this needs:

1. Adding it to the next assets update
2. Adding a mapping so is linked with the rest

[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/data/global.db:1]: > Adding it to the next assets update

I don't think that it's needed because it is a constant asset and thus will be searched in the pacakged db

> Adding a mapping so is linked with the rest

Yes, will do

> Can you also populate the started field? I believe you already had this constant somewhere

Will do

[the remark this answers is not in the document]
Comment (nebolax) [reply rotkehlchen/data/global.db:1]: Done.

New globaldb diff
```sql
INSERT INTO assets(rowid,identifier,name,type) VALUES(5075,'MATIC','MATIC','B');
INSERT INTO common_asset_details(rowid,identifier,symbol,coingecko,cryptocompare,forked,started,swapped_for) VALUES(5075,'MATIC','MATIC','matic-network','MATIC',NULL,1590824836,NULL);
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(11,'polygon etherscan','',0,1,'0.25','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(12,'ankr','https://rpc.ankr.com/polygon',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(13,'BlockPi','https://polygon.blockpi.network/v1/rpc/public',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(14,'PublicNode','https://polygon-bor.publicnode.com',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(15,'DefiLlama','https://polygon.llamarpc.com',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(16,'1rpc','https://1rpc.io/matic',0,1,'0.15','POLYGON_POS');
```

Assets update PR: https://github.com/rotki/assets/pull/111

File: rotkehlchen/api/rest.py [hunk 4]
```diff
@@ -654,7 +654,7 @@ def get_supported_chains(self) -> Response:
             if blockchain == SupportedBlockchain.OPTIMISM:
                 data['native_asset'] = A_ETH.serialize()
             elif blockchain == SupportedBlockchain.POLYGON_POS:
-                data['native_asset'] = A_POLYGON_POS_MATIC.serialize()
+                data['native_asset'] = A_MATIC.serialize()
```
Comment (LefterisJP) [inline rotkehlchen/api/rest.py:657]: Why this change? Shouldn't they be equivalent? And since POLYGON_POS_MATIC is the asset in the POS chain that should be the correct one.

File: rotkehlchen/chain/evm/node_inquirer.py [hunk 5]
```diff
@@ -283,7 +286,7 @@ def get_multi_balance(
             accounts: Sequence[ChecksumEvmAddress],
             call_order: Optional[Sequence[WeightedNode]] = None,
     ) -> dict[ChecksumEvmAddress, FVal]:
-        """Returns a dict with keys being accounts and balances in ETH
+        """Returns a dict with keys being accounts and balances in chain native token.
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/node_inquirer.py:289]:
```suggestion
        """Returns a dict with keys being accounts and balances in the chain native token.
```

File: rotkehlchen/chain/evm/node_inquirer.py [hunk 6]
```diff
@@ -207,6 +209,7 @@ def __init__(
         self.rpc_timeout = rpc_timeout
         self.chain_id: SUPPORTED_CHAIN_IDS = blockchain.to_chain_id()  # type: ignore[assignment]
         self.chain_name = self.blockchain.name.lower()
+        self.native_asset = native_asset
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/node_inquirer.py:212]: Here and elsewhere please be consistent. Call it token. In the same file you end up using different terms for the same thing.

```suggestion
        self.native_token = native_asset
```

**Review by LefterisJP — COMMENTED** [review]
Something is wrong with the VCR repo as the failing test shows.

Also left some important comments. I am not sure at all about this Matic asset approach

File: rotkehlchen/api/rest.py (hunk 4 above)
Comment (nebolax) [reply rotkehlchen/api/rest.py:657]: I thought about it and yeah, it sounds reasonable. Will change

**Comment by nebolax** [comment]
New globaldb diff:
```
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(11,'polygon etherscan','',0,1,'0.25','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(12,'ankr','https://rpc.ankr.com/polygon',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(13,'BlockPi','https://polygon.blockpi.network/v1/rpc/public',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(14,'PublicNode','https://polygon-bor.publicnode.com',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(15,'DefiLlama','https://polygon.llamarpc.com',0,1,'0.15','POLYGON_POS');
INSERT INTO default_rpc_nodes(identifier,name,endpoint,owned,active,weight,blockchain) VALUES(16,'1rpc','https://1rpc.io/matic',0,1,'0.15','POLYGON_POS');
```

Matic token is already in the db and is a part of matic asset colletion.

**Review by LefterisJP — APPROVED** [review]
LGTM. Uniswap subgraph lead to tests failure. Other than that small nitpick for later. Merging.
